rasa/core/tracker_store.py

```diff
@@ -1236,3 +1234,132 @@ def _load_from_module_name_in_endpoint_config(
             f"Using `InMemoryTrackerStore` instead."
         )
         return InMemoryTrackerStore(domain)
+
+
+class AwaitableTrackerStore(TrackerStore):
+    """Wraps a tracker store so it can be implemented with async overrides."""
+
+    def __init__(
+            self,
+            tracker_store: TrackerStore,
+    ) -> None:
+        """Create a `AwaitableTrackerStore`.
+        Args:
+            tracker_store: the wrapped tracker store.
+        """
+
+        self._tracker_store = tracker_store
+
+        super().__init__(tracker_store.domain, tracker_store.event_broker)
+
+    @property
+    def domain(self) -> Optional[Domain]:
+        return self._tracker_store.domain
+
+    @domain.setter
+    def domain(self, domain: Optional[Domain]) -> None:
+        self._tracker_store.domain = domain
+
+    @staticmethod
+    def create(obj: Union["TrackerStore", EndpointConfig, None],
+               domain: Optional[Domain] = None,
+               event_broker: Optional[EventBroker] = None, ) -> Optional[TrackerStore]:
+        if isinstance(obj, TrackerStore):
+            return AwaitableTrackerStore(obj)
+        else:
+            return None
+
+    def on_tracker_store_error(self, error: Exception) -> None:
+        logger.error(
+            f"Error happened when trying to save conversation tracker to "
+            f"'{self._tracker_store.__class__.__name__}'. Falling back to use "
+            f"the '{InMemoryTrackerStore.__name__}'. Please "
+            f"investigate the following error: {error}."
+        )
+
+    async def get_or_create_tracker(
```

Comment:
Do we have to re-implement these methods that are already on `TrackerStore`?

Reply:
Ah yes thank you! Those were residual - removed the duplicated methods